worlds/ahit/Rules.py

```diff
@@ -0,0 +1,988 @@
+
+            continue
+
+        add_rule(loc, lambda state: can_use_hookshot(state, world))
+
+    dummy_entrances: List[Entrance] = []
+      
+    for (key, acts) in act_connections.items():
+        if "Arctic Cruise" in key and not world.is_dlc1():
+            continue
+
+        i: int = 1
+        entrance: Entrance = world.multiworld.get_entrance(key, world.player)
+        region: Region = entrance.connected_region
+        access_rules: List[Callable[[CollectionState], bool]] = []
+        dummy_entrances.append(entrance)
+
+        # Entrances to this act that we have to set access_rules on
+        entrances: List[Entrance] = []
+
+        for act in acts:
+            act_entrance: Entrance = world.multiworld.get_entrance(act, world.player)
+            access_rules.append(act_entrance.access_rule)
+            required_region = act_entrance.connected_region
+            name: str = f"{key}: Connection {i}"
+            new_entrance: Entrance = connect_regions(required_region, region, name, world.player)
+            entrances.append(new_entrance)
+
+            # Copy access rules from act completions
+            if "Free Roam" not in required_region.name:
+                rule: Callable[[CollectionState], bool]
+                name = f"Act Completion ({required_region.name})"
+                rule = world.multiworld.get_location(name, world.player).access_rule
+                access_rules.append(rule)
+
+            i += 1
+
+        for e in entrances:
+            for rules in access_rules:
+                add_rule(e, rules)
+
+    for e in dummy_entrances:
+        set_rule(e, lambda state: False)
+
+    set_event_rules(world)
+
+    if world.options.EndGoal.value == 1:
+        world.multiworld.completion_condition[world.player] = lambda state: state.has("Time Piece Cluster", world.player)
+    elif world.options.EndGoal.value == 2:
+        world.multiworld.completion_condition[world.player] = lambda state: state.has("Rush Hour Cleared", world.player)
+
+
+def set_specific_rules(world: "HatInTimeWorld"):
+    add_rule(world.multiworld.get_location("Mafia Boss Shop Item", world.player),
+             lambda state: state.has("Time Piece", world.player, 12)
+             and state.has("Time Piece", world.player, world.get_chapter_cost(ChapterIndex.BIRDS)))
+
+    add_rule(world.multiworld.get_location("Spaceship - Rumbi Abuse", world.player),
+             lambda state: state.has("Time Piece", world.player, 4))
+
+    set_mafia_town_rules(world)
+    set_botb_rules(world)
+    set_subcon_rules(world)
+    set_alps_rules(world)
+
+    if world.is_dlc1():
+        set_dlc1_rules(world)
+
+    if world.is_dlc2():
+        set_dlc2_rules(world)
+
+    difficulty: Difficulty = get_difficulty(world)
+
+    if difficulty >= Difficulty.MODERATE:
+        set_moderate_rules(world)
+
+    if difficulty >= Difficulty.HARD:
+        set_hard_rules(world)
+
+    if difficulty >= 2:
```

Comment:
```suggestion
    if difficulty >= Difficulty.EXPERT:
```
just for consistency